src/documents.rs

```diff
@@ -38,29 +39,30 @@ use serde::{de::DeserializeOwned, Deserialize, Serialize};
 ///     genres: Vec<String>,
 /// }
 ///
-/// async fn usage(client: Client) {
+/// async fn usage(client: Client<IsahcClient>) {
 ///     // Default settings with the distinct, searchable, displayed, filterable, and sortable fields set correctly.
 ///     let settings: Settings = Movie::generate_settings();
 ///     // Index created with the name `movie` and the primary key set to `movie_id`
-///     let index: Index = Movie::generate_index(&client).await.unwrap();
+///     let index: Index<IsahcClient> = Movie::generate_index(&client).await.unwrap();
 /// }
 /// ```
 pub use meilisearch_index_setting_macro::IndexConfig;
 
+use crate::request::HttpClient;
 use crate::settings::Settings;
 use crate::tasks::Task;
 use crate::Client;
 use crate::{errors::Error, indexes::Index};
 
-#[async_trait]
-pub trait IndexConfig {
+#[async_trait(?Send)]
```

Comment:
I needed it to build with wasm32.
But this might not be a very good way to write it.